### PR TITLE
Report package test fixes part 3

### PR DESCRIPTION
**Pull request by holderb5**
## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by kevinhinterlong** [comment]
The [lint step](https://cd.screwdriver.cd/pipelines/6102/builds/667252/steps/test-navi-reports) is failing

File: packages/reports/tests/acceptance/multi-datasource-test.js [hunk 1]
```diff
@@ -45,9 +45,9 @@ module('Acceptance | multi-datasource report builder', function(hooks) {
 
     await selectChoose('.filter-builder__select-trigger', 'Between');
     await clickTrigger('.filter-values--dimension-date-range-input__low-value .ember-basic-dropdown-trigger');
-    await click($('button.ember-power-calendar-day--current-month:contains(4)')[0]);
+    await click('.ember-power-calendar-day--current-month:contains(4)');
```
Comment (jkusa) [inline packages/reports/tests/acceptance/multi-datasource-test.js:48]: Does this selector work without jquery?

Comment (holderb5) [reply packages/reports/tests/acceptance/multi-datasource-test.js:48]: Forgot to double check that, it doesnt work, what would you suggest? we were getting an error when using jquery in this file

File: packages/reports/tests/integration/components/print-report-view-test.js [hunk 2]
```diff
@@ -1,131 +1,149 @@
...
-                    metrics: [
-                      {
-                        metric: 'adClicks',
-                        parameters: {},
-                        canonicalName: 'adClicks',
-                        longName: 'Ad Clicks'
-                      }
-                    ]
-                  }
+    this.set('report', {
```
Comment (kevinhinterlong) [inline packages/reports/tests/integration/components/print-report-view-test.js:66]: Why is this no longer doing `store.createRecord`?

File: packages/reports/tests/integration/components/print-report-view-test.js [hunk 3]
```diff
@@ -1,131 +1,149 @@
...
+        limit: null,
+        requestVersion: '2.0',
+        filters: [
+          {
+            type: 'timeDimension',
+            dataSource: 'bardOne',
+            field: 'network.dateTime',
+            parameters: { grain: 'day' },
+            operator: 'bet',
+            values: ['11-04-2020', '11-06-2020']
```
Comment (kevinhinterlong) [inline packages/reports/tests/integration/components/print-report-view-test.js:106]: I think these should line up with the mock response
```suggestion
            values: ['2015-11-09', '2015-11-14']
```

File: packages/reports/tests/integration/components/report-actions/export-test.js [hunk 4]
```diff
@@ -55,15 +52,31 @@ module('Integration | Component | report actions - export', function(hooks) {
...
+        limit: null,
+        requestVersion: '2.0',
+        filters: [
+          {
+            type: 'timeDimension',
+            dataSource: 'bardOne',
+            field: 'network.dateTime',
+            parameters: { grain: 'day' },
+            operator: 'bet',
+            values: ['11-04-2020', '11-06-2020']
```
Comment (kevinhinterlong) [inline packages/reports/tests/integration/components/report-actions/export-test.js:66]: Let's use the same dates as before
```suggestion
            values: ['2015-10-02', '2015-10-14']
```

File: packages/reports/tests/integration/components/report-actions/export-test.js [hunk 5]
```diff
@@ -55,15 +52,31 @@ module('Integration | Component | report actions - export', function(hooks) {
 
     run(() => {
```
Comment (kevinhinterlong) [inline packages/reports/tests/integration/components/report-actions/export-test.js:53]: The `run()` is probably not needed

File: packages/reports/tests/integration/components/report-actions/export-test.js [hunk 6]
```diff
@@ -55,15 +52,31 @@ module('Integration | Component | report actions - export', function(hooks) {
...
-          })
-        ]
+        table: 'network',
+        dataSource: 'bardOne',
+        limit: null,
+        requestVersion: '2.0',
+        filters: [
+          {
+            type: 'timeDimension',
+            dataSource: 'bardOne',
```
Comment (kevinhinterlong) [inline packages/reports/tests/integration/components/report-actions/export-test.js:62]:
```suggestion
            source: 'bardOne',
```

File: packages/reports/addon/components/filter-values/null-input.ts [hunk 7]
```diff
@@ -8,6 +8,9 @@ import Args from './args-interface';
 export default class NullInput extends Component<Args> {
   constructor(owner: unknown, args: Args) {
     super(owner, args);
-    this.args.onUpdateFilter({ values: [true] });
+    const { values } = this.args.filter;
```
Comment (kevinhinterlong) [inline packages/reports/addon/components/filter-values/null-input.ts:11]: Rebase this to use my changes

File: packages/reports/tests/integration/components/print-report-view-test.js [hunk 8]
```diff
@@ -1,131 +1,149 @@
...
+      },
+      request: {
+        table: 'network',
+        dataSource: 'bardOne',
+        limit: null,
+        requestVersion: '2.0',
+        filters: [
+          {
+            type: 'timeDimension',
+            dataSource: 'bardOne',
```
Comment (kevinhinterlong) [inline packages/reports/tests/integration/components/print-report-view-test.js:102]:
```suggestion
            source: 'bardOne',
```

File: packages/reports/tests/integration/components/report-view-test.js [hunk 9]
```diff
@@ -79,17 +84,44 @@ module('Integration | Component | report view', function(hooks) {
...
+          limit: null,
+          requestVersion: '2.0',
+          filters: [
+            {
+              type: 'timeDimension',
+              dataSource: 'bardOne',
+              field: 'network.dateTime',
+              parameters: { grain: 'day' },
+              operator: 'bet',
+              values: ['11-04-2020', '11-06-2020']
```
Comment (kevinhinterlong) [inline packages/reports/tests/integration/components/report-view-test.js:99]:
```suggestion
              values: ['2015-11-09', '2015-11-14']
```

File: packages/reports/tests/integration/components/report-view-test.js [hunk 10]
```diff
@@ -79,17 +84,44 @@ module('Integration | Component | report view', function(hooks) {
...
-        }),
+        request: {
+          table: 'network',
+          dataSource: 'bardOne',
+          limit: null,
+          requestVersion: '2.0',
+          filters: [
+            {
+              type: 'timeDimension',
+              dataSource: 'bardOne',
```
Comment (kevinhinterlong) [inline packages/reports/tests/integration/components/report-view-test.js:95]:
```suggestion
              source: 'bardOne',
```

File: packages/reports/tests/integration/components/report-view-test.js [hunk 11]
```diff
@@ -98,9 +130,7 @@ module('Integration | Component | report view', function(hooks) {
                   config: {
```
Comment (kevinhinterlong) [inline packages/reports/tests/integration/components/report-view-test.js:130]: This should be `config: {}` for a metric series

File: packages/reports/tests/integration/components/report-view-test.js [hunk 12]
```diff
@@ -117,30 +147,38 @@ module('Integration | Component | report view', function(hooks) {
...
+          type: 'timeDimension',
+          dataSource: 'bardOne',
+          field: 'network.dateTime',
+          parameters: { grain: 'day' },
+          operator: 'bet',
+          values: ['11-04-2020', '11-06-2020']
+        }
+      ],
+      columns: [
+        {
```
Comment (kevinhinterlong) [inline packages/reports/tests/integration/components/report-view-test.js:165]:
```suggestion
        {
           type: 'timeDimension',
           source: 'bardOne',
           field: 'network.dateTime',
           parameters: { grain: 'day' },
        },
        {
```

File: packages/reports/tests/integration/components/report-view-test.js [hunk 13]
```diff
@@ -117,30 +147,38 @@ module('Integration | Component | report view', function(hooks) {
...
-      filters: [],
-      sort: [
+      table: 'network',
+      dataSource: 'bardOne',
+      limit: null,
+      requestVersion: '2.0',
+      filters: [
+        {
+          type: 'timeDimension',
+          dataSource: 'bardOne',
```
Comment (kevinhinterlong) [inline packages/reports/tests/integration/components/report-view-test.js:157]:
```suggestion
          source: 'bardOne',
```

File: packages/reports/tests/integration/components/report-view-test.js [hunk 14]
```diff
@@ -117,30 +147,38 @@ module('Integration | Component | report view', function(hooks) {
...
+          field: 'network.dateTime',
+          parameters: { grain: 'day' },
+          operator: 'bet',
+          values: ['11-04-2020', '11-06-2020']
```
Comment (kevinhinterlong) [inline packages/reports/tests/integration/components/report-view-test.js:161]:
```suggestion
          values: ['2015-11-09', '2015-11-14']
```

File: packages/reports/tests/integration/components/report-view-test.js [hunk 15]
```diff
@@ -163,20 +196,13 @@ module('Integration | Component | report view', function(hooks) {
...
-            attributes: { name: 'adClicks' },
-            type: 'metric',
-            displayName: 'Ad Clicks'
-          }
-        ]
+        columnAttributes: {
+          c1: { canAggregateSubtotal: false },
+          c2: { canAggregateSubtotal: false }
+        },
+        showTotals: []
```
Comment (kevinhinterlong) [inline packages/reports/tests/integration/components/report-view-test.js:205]:
```suggestion
        }
```

**Review by kevinhinterlong — COMMENTED** [review]
Little things

File: packages/reports/tests/integration/components/print-report-view-test.js [hunk 16]
```diff
@@ -1,131 +1,149 @@
...
+            area: false,
+            stacked: false
+          },
+          axis: {
+            y: {
+              series: {
+                type: 'dateTime',
+                config: {
+                  timeGrain: 'day',
+                  metricCid: 'c2'
```
[the remark this answers is not in the document]
Comment (holderb5) [reply packages/reports/tests/integration/components/print-report-view-test.js:87]: So its just an empty config like: config: {}?

[the remark this answers is not in the document]
Comment (kevinhinterlong) [reply packages/reports/tests/integration/components/print-report-view-test.js:87]: My comment also changes it to be a metric config^ which is why it's empty

File: packages/reports/tests/integration/components/report-view-test.js (hunk 12 above)
Comment (holderb5) [reply packages/reports/tests/integration/components/report-view-test.js:165]: When adding the timedimension in we lose the metric label capability thats required for this test

File: packages/reports/tests/integration/components/print-report-view-test.js (hunk 2 above)
Comment (holderb5) [reply packages/reports/tests/integration/components/print-report-view-test.js:66]: When keeping the same format of just defining the request and visualization the render failed. I could be messing something up though.

File: packages/reports/tests/integration/components/print-report-view-test.js (hunk 16 above)
[the remark this answers is not in the document]
Comment (holderb5) [reply packages/reports/tests/integration/components/print-report-view-test.js:87]: With the change of series the tests fail, even when changed to type: 'metric', config: {}

File: packages/reports/tests/integration/components/report-view-test.js (hunk 12 above)
Comment (kevinhinterlong) [reply packages/reports/tests/integration/components/report-view-test.js:161]: Ah ok then it looks like you also need to update the filter values to be what they were before `['current', 'next']`

File: packages/reports/tests/integration/components/report-actions/export-test.js [hunk 17]
```diff
@@ -50,37 +49,6 @@ module('Integration | Component | report actions - export', function(hooks) {
   test('Component is not disabled for unsaved reports', async function(assert) {
     assert.expect(1);
 
-    run(() => {
-      let request = {
```
Comment (kevinhinterlong) [inline packages/reports/tests/integration/components/report-actions/export-test.js:57]: Put this part back you just don't need the `run` wrapper

File: packages/reports/tests/integration/components/report-actions/export-test.js [hunk 18]
```diff
@@ -50,37 +49,6 @@ module('Integration | Component | report actions - export', function(hooks) {
...
-        table: 'network',
-        dataSource: 'bardOne',
-        limit: null,
-        requestVersion: '2.0',
-        filters: [
-          {
-            type: 'timeDimension',
-            dataSource: 'bardOne',
```
Comment (kevinhinterlong) [inline packages/reports/tests/integration/components/report-actions/export-test.js]: same updates for `source`/`values` here

File: packages/reports/tests/integration/components/report-view-test.js [hunk 19]
```diff
@@ -79,30 +84,51 @@ module('Integration | Component | report view', function(hooks) {
...
                   type: 'metric',
                   config: {
-                    metrics: [
-                      {
-                        metric: 'adClicks',
-                        parameters: {},
-                        canonicalName: 'adClicks'
-                      }
-                    ]
+                    metrics: [{}]
```
Comment (kevinhinterlong) [inline packages/reports/tests/integration/components/report-view-test.js:131]:
```suggestion
```

**Review by kevinhinterlong — APPROVED** [review]

**Review by jkusa — APPROVED** [review]

**Pull request merged**